derive-input/src/internal/table.rs

```diff
@@ -5,20 +5,33 @@ use crate::api::{
 };
 use quote::format_ident;
 use spacetime_bindings_macro_input::table::{ColumnArgs, TableArgs};
-use syn::DeriveInput;
+use spacetime_bindings_macro_input::{match_meta, util::check_duplicate};
+use syn::{DeriveInput, Ident, meta::parser, parse::Parser};
 
 pub(in crate::internal) fn try_parse(
     args: proc_macro2::TokenStream,
     input: &DeriveInput,
     table_args: &TableArgs,
     column_args: &ColumnArgs<'_>,
+    plural_name: Option<syn::Ident>,
 ) -> syn::Result<Table> {
     let rust_struct = crate::internal::rust::table::map_struct(&input);
 
     let spacetimedb_table = SpacetimeDBTable::map(table_args);
 
+    // Use provided plural_name or parse it from args if not provided
+    let plural_name = if let Some(name) = plural_name {
+        name
+    } else {
+        parse_plural_name_from_args(&args)?
+            .ok_or_else(|| syn::Error::new(
+                proc_macro2::Span::call_site(),
+                format_args!("PluralName must be set in `#[dsl(plural_name = PluralName)]`, e.g. `plural_name = {}s`.", spacetimedb_table.singular_name),
+            ))?
+    };
+
```

Comment:
Remove, the parse_plural_name_from_args function should already return the error when it's executed in line 18 of the derive-input/src/internal.rs file.